lib/pg_ha_migrations/safe_statements.rb

```diff
@@ -582,16 +520,35 @@ def exec_migration(conn, direction)
   end
 
   def safely_acquire_lock_for_table(table, &block)
+    nested_lock = Thread.current[__method__]
+
     _check_postgres_adapter!
-    table = table.to_s
-    quoted_table_name = connection.quote_table_name(table)
+
+    table = PgHaMigrations::Struct::Table.from_table_name(table)
+
+    # Disallow nested locks unless targeting the same table
+    if nested_lock && nested_lock != table
```

Comment:
Now that this is the actual value we can simply:
```suggestion
    if nested_lock != table
```

Reply:
I don't think so. In the case where there isn't a nested lock that would turn into `nil != table` which would evaluate to true and we'd incorrectly raise an error

Reply:
Ah, I see that now.